tests/st/calicoctl/test_ipip.py

```diff
@@ -130,3 +203,53 @@ def remove_tunl_ip(self):
             output = subprocess.check_output(["ip", "addr", "del", ipnet, "dev", "tunl0"])
         except subprocess.CalledProcessError:
             return
+
+    def modify_subnet(self, host, prefixlen):
+        """
+        Update the calico node resource to use the specified prefix length.
+
+        Returns the current mask size.
+        """
+        noder = json.loads(host.calicoctl(
+            "get node %s --output=json" % host.get_hostname()))
+        assert len(noder) == 1
+
+        # Get the current network and prefix len
+        ipnet = IPNetwork(noder[0]["spec"]["bgp"]["ipv4Address"])
+        current_prefix_len = ipnet.prefixlen
+
+        # Update the prefix length
+        ipnet.prefixlen = prefixlen
+        noder[0]["spec"]["bgp"]["ipv4Address"] = str(ipnet)
+
+        # Write the data back again.
+        host.writejson("new_data", noder)
+        host.calicoctl("apply -f new_data")
+        return current_prefix_len
+
+    def assert_ipip_routing(self, host1, workload_host1, workload_host2, expect_ipip):
+        """
+        Test whether IPIP is being used as expected on host1 when pinging workload_host2
+        from workload_host1.
+        """
+        def check():
+            orig_tx = self.get_tunl_tx(host1)
+            workload_host1.execute("ping -c 2 -W 1 %s" % workload_host2.ip)
+            if expect_ipip:
+                assert self.get_tunl_tx(host1) == orig_tx + 2
+            else:
+                assert self.get_tunl_tx(host1) == orig_tx
```

Comment:
Where is this function asserting if the ping succeeds when `expect_ipip` is `False`? Only assertion being made here is tunl TX count is the same which would be the case even if ping doesn't work.